core/src/main/java/org/apache/gravitino/storage/relational/service/NameIdMappingService.java

```diff
@@ -0,0 +1,156 @@
+ * under the License.
+ */
+package org.apache.gravitino.storage.relational.service;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.Scheduler;
+import com.google.common.base.Objects;
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import org.apache.gravitino.Entity.EntityType;
+import org.apache.gravitino.NameIdentifier;
+
+public class NameIdMappingService implements Closeable {
+
+  private static volatile NameIdMappingService instance;
+
+  private Cache<EntityIdentifier, Long> ident2IdCache;
+
+  private NameIdMappingService() {
+    this.ident2IdCache =
+        Caffeine.newBuilder()
+            .expireAfterAccess(24 * 3600 * 1000L /* 1 day */, TimeUnit.MILLISECONDS)
+            .maximumSize(1000000)
+            .initialCapacity(1000)
+            .scheduler(
+                Scheduler.forScheduledExecutorService(
+                    new ScheduledThreadPoolExecutor(
+                        1,
+                        new ThreadFactoryBuilder()
+                            .setDaemon(true)
+                            .setNameFormat("ident-to-id-cleaner-%d")
+                            .build())))
+            .build();
+  }
+
+  public static class EntityIdentifier {
+    final NameIdentifier ident;
+    final EntityType type;
+
+    private EntityIdentifier(NameIdentifier ident, EntityType type) {
+      this.ident = ident;
+      this.type = type;
+    }
+
+    public static EntityIdentifier of(NameIdentifier ident, EntityType type) {
+      return new EntityIdentifier(ident, type);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (!(o instanceof EntityIdentifier)) {
+        return false;
+      }
+      EntityIdentifier that = (EntityIdentifier) o;
+      return Objects.equal(ident, that.ident) && type == that.type;
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hashCode(ident, type);
+    }
+  }
+
+  public static NameIdMappingService getInstance() {
+    if (instance == null) {
+      synchronized (NameIdMappingService.class) {
+        if (instance == null) {
+          instance = new NameIdMappingService();
+        }
+      }
+    }
+
+    return instance;
+  }
+
+  public void put(EntityIdentifier key, Long value) {
+    ident2IdCache.put(key, value);
+  }
+
+  public Long get(EntityIdentifier key, Function<EntityIdentifier, Long> mappingFunction) {
+    return ident2IdCache.get(key, mappingFunction);
+  }
+
+  public Long get(EntityIdentifier key) {
+    return ident2IdCache.getIfPresent(key);
```

Comment:
If the cache is not hit, does the upper layer need to retrieve it from the underlying storage by itself?

Reply:
Yes, in some scenarios we only need to judge whether the value associated with the key exists.

Reply:
I am a bit confused. Shouldn't the cacheService automatically help the caller to query the underlying layer when the cache expires? But I didn't see the logic of querying from the underlying storage in your cache.

Reply:
You can use  
```
  public Long get(EntityIdentifier key, Function<EntityIdentifier, Long> mappingFunction) {
    return ident2IdCache.get(key, mappingFunction);
  }
```
instead.

Reply:
Your class name is NameIdMappingService, not a caching service. Caching seems to be just an optimization technique in your service, so I think the cache should be transparent to the caller.

Reply:
As your current implementation, ` Long get(EntityIdentifier key)`  makes the caller feel that they can directly obtain the `id` through the `key`, but in fact it only performs a cache query. That's my point.